### DISCUSSION TITLE
💻你的電腦不夠強嗎?別怕~有VDI

### DISCUSSION THREAD

**Discussion by JunTingLin** [1101-GS4719-C/Python-Community-GS4719-C#409] [General]
# 💻你的電腦不夠強嗎?別怕~有VDI

## 甚麼是虛擬桌面基礎架構 (VDI)❓
虛擬桌面基礎架構 (VDI) 是使用虛擬機器提供和管理虛擬桌面的虛擬化解決方案。VDI 會將桌面環境託管在一個集中式伺服器上，並應要求部署給終端使用者。使用端點裝置（筆記型電腦、平板電腦等）即可透過網路存取。💫

有了 VDI 解決方案，組織就能實現許多效益。桌面運算是在主機伺服器上進行，而非在端點裝置上進行，因此端點的硬體需求較低。

## 我們學校也有喔
中央大學去年2月開始建置整合AI教學雲、3D雲、軟體雲的[「多雲整合虛擬桌面平台」](https://vdi.ncu.edu.tw/start/index.html)，在這一波防疫期間也正好派上用場，授權的軟體包括微軟Office 2019、Acrobat 2017、3D繪圖雲、統計分析雲、會聲會影、AI雲等虛擬桌面，提供師生「隨時、隨地、隨需」使用，具有「免下載、免安裝、免設定、免更新」的特色，學生不需花費心思在軟體安裝和電腦環境設定，就可有效率地進行研究與學習。🚴‍♂️
![螢幕擷取畫面 2022-03-06 015657](https://user-images.githubusercontent.com/92431095/156894812-d5e6095c-fa40-4827-aaa5-555d4c2f7844.png)

## 我用這個要幹嘛
1. 上面也有提到，客戶端(clinet)透過網路的方式連到集中的伺服器上，所以本身電腦配置不用太高
2. 學校提供不同的虛擬桌面，分別慣了各自的授權軟體，比方剪輯雲的Adobe Premiere Pro

## 如何使用學校虛擬桌面
1. 直接透過瀏覽器使用
![螢幕擷取畫面 2022-03-06 014800](https://user-images.githubusercontent.com/92431095/156894864-3d1cc062-5a07-4ced-ba38-f3aa62e432ac.png)

2. 下載VM horizon
![螢幕擷取畫面 2022-03-06 015032](https://user-images.githubusercontent.com/92431095/156894831-42976dce-bd2a-493e-bae1-3bcfc915f80c.png)

## 結語
python往更高階發展可能就是機器學習、AI人工智慧、圖像識別等等，鄉性都需要強大的GPU去計算，學校某些雲支持GPU加速，相信可以派上用場。

## 參考
[探 VDI 虛擬桌面基礎結構](https://www.infortrend.com/tw/solutions/vdi)

[中央大學-多雲整合虛擬桌面平台](https://vdi.ncu.edu.tw/start/index.html)

**Comment by ghost** [comment]
這種情況我一律建議砸錢
顯卡多香>///<
(沒，先看看錢包君願不願意...

**Comment by pekoranoinu** [comment]
沒$$的我也能享受到真是開心。

**Comment by xkllkx** [comment]
有人有嘗試過使用這個來訓練模型嗎？

**Reply by gnsJhenJie** [reply]
一般訓練模型時間常常超過兩小時，但VDI有限制單次使用最多兩小時，所以要在VDI上訓練模型可能比較困難。不過，電算中心也有提供GPU伺服器的長期借用(單次最多三個月)，可以參考https://www.cc.ncu.edu.tw/storage/pages/4/NCU_GPU%E4%BC%BA%E6%9C%8D%E5%99%A8%E5%80%9F%E7%94%A8%E7%94%B3%E8%AB%8B%E5%96%AE_v112.pdf

**Comment by bloodtrail0511** [comment]
這樣可以運用有限資源，利用遠端伺服器克服計算資源不足的問題，對需要大量圖像或資料處理的工作好處多多

**Comment by PoooYen** [comment]
太用心了拉，感謝教學

**Comment by gnsJhenJie** [comment]
VDI每兩小時會中斷使用者的連線，儲存在VDI上的資料也會被刪除，如果有資料保留需求可以透過Horizon Client 掛載自己電腦上的儲存空間至VDI上([教學文件](https://drive.google.com/file/d/1k4-n5yKttwCicih3m7C5L2Sv3B_96uyO/view))

**Comment by yucheng1028** [comment]
感覺還不錯，但祈禱我的1060不要那麼快被淘汰，讓我只能用這個2ㄏ

**Comment by Ku-Hao-Chen** [comment]
謝謝分享，感謝教學。

**Comment by hapiwangy** [comment]
原來學校也有提供這個服務，真的是太方便了

**Comment by HaoFish123** [comment]
之前有發現，但沒有摸索，所以謝謝你的教學。

**Comment by cordydeng** [comment]
原來學校還有提供這麼完善的服務阿，謝謝分享

**Comment by Yuhsin000** [comment]
以後用用看，省錢

**Comment by jason79461385** [comment]
之前有做到需要跑大量圖片的專題，如果早點知道能用這個就不會那麼痛苦了呢

**Comment by stan890425** [comment]
那這樣對網路的需求會不會很高

**Comment by jcpan821** [comment]
好酷喔，改天來用看看

**Comment by alexthebigpp** [comment]
感謝分享，非常有趣

**Comment by minniewei** [comment]
感謝大大分享相關的資訊

**Comment by Ben6uo** [comment]
哇 太神了 我們竟然有這種福利

**Comment by yenwei12138** [comment]
感謝分享，這在未來需要在學校訓練模型時感覺用的上

**Comment by etin2468** [comment]
感謝分享，這樣我的破電腦也有救了

**Comment by Lynn12378** [comment]
感謝分享，以後有需要開發專題就可以用了
